app/main.py

```diff
@@ -1,3 +1,19 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable, Any
+
+
+def cache(func: Callable) -> Callable:
+    cache_dict = {}
+
+    def inner(*args) -> Any:
+        nonlocal cache_dict
```

Comment:
Nonlocal is unneeded here, read more about it by the [link](https://www.geeksforgeeks.org/python-nonlocal-keyword/)